services/cd-service/pkg/repository/transformer.go

```diff
@@ -658,6 +663,28 @@ func (c *CreateApplicationVersion) Transform(
 	return fmt.Sprintf("created version %d of %q", version, c.Application), nil
 }
 
+func validateDownstreamEnvs(downstreamEnvs []types.EnvName, sortedEnvs []types.EnvName, configs map[types.EnvName]config.EnvironmentConfig) error {
+	notDownstreamEnvs := []types.EnvName{}
```

Comment:
hm, maybe call this something else?
```suggestion
	upstreamLatestEnvs := []types.EnvName{}
```